R/stack.R

```diff
@@ -582,30 +583,64 @@ getDotVars <- function(env) {
 #' 
 #' Gets a representation of a promise without forcing the promise
 #' 
+#' @aliases .vsc.promise
 #' @param name The name of the variable
 #' @param env The environment in which to evaluate
-#' @return A named list containing the expression that will be evaluated, the environment in which it will be evaluated and a string representation of the expression
+#' @return An object of class \code{".vsc.promise"}; a named list containing the expression that will be evaluated, the status whether the promise has already been evaluated, the value if it has already been evaluated, and the environment in which the unevaluated promise will be evaluated. 
+#' @examples 
+#' ## create a promise
+#' e <- new.env()
+#' delayedAssign("x", {message("evaluating promise..."); 1L}, assign.env = e)
+#' vscDebugger:::getPromiseVar("x", e)
+#' 
+#' ## evaluate it...
+#' print(e$x)
+#' 
+#' ## is it still a promise? (It depends...)
+#' stopifnot(vscDebugger:::isPromise("x", e, strict = FALSE))
+#' stopifnot(!vscDebugger:::isPromise("x", e, strict = TRUE))
+#' 
+#' ## get info again
+#' vscDebugger:::getPromiseVar("x", e)
 #' 
 getPromiseVar <- function(name, env) {
-  promiseExpr <- pryr:::promise_code(name, env)
-  promiseCode <- try(paste0(toString(promiseExpr), collapse = ';'), silent=getOption('vsc.trySilent', default=TRUE))
-  if (class(promiseCode) == 'try-error') promiseCode <- '???'
-  promiseEnv <- pryr:::promise_env(name, env)
-  var <- list(
-    promiseCode = promiseCode,
-    promiseEnv = promiseEnv,
-    promiseExpr = promiseExpr
+  structure(
+    getPromiseInfo(name, env),
+    class = c(".vsc.promise", ".vsc.InternalClass")
   )
-  class(var) <- c('.vsc.promise', '.vsc.InternalClass')
-  return(var)
 }
 
-
-isPromise <- function(name, env) {
-  if (pryr:::is_promise2(name, env)) {
-    return(!pryr:::promise_evaled(name, env))
-  }
-  return(FALSE)
+#' Get information about a promise
+#' 
+#' @param name The name of the variable
+#' @param env The environment in which to evaluate
+#' @return A named list: 
+#' \itemize{
+#'   \item{\code{code}: }{the expression that will be evaluated}
+#'   \item{\code{environment}: }{the environment where the promise is evaluated}
+#'   \item{\code{evaluated}: }{logical flag if the promise has been already evaluated}
+#'   \item{\code{value}: }{optional node; the value of the evaluated promise}
+#' }
+#' @keywords internal
+#' @useDynLib vscDebugger c_promise_info 
+getPromiseInfo <- function(name, env) {
+  sym <- as.name(name)
+  .Call(c_promise_info, sym, env)
+}
+
+#' Test if in object is a promise
+#' 
+#' @param name [character(1L)] the name of the object
+#' @param env [environment] the environment of the object 
+#' @param strict [logical(1L)] if \code{strict} is \code{TRUE} 
```

Comment:
When installing the package I get the following warnings:
```
Rd warning: C:/Users/Manuel/AppData/Local/Temp/RtmpyM0oij/R.INSTALLa6cc51826981/vscDebugger/man/isPromise.Rd:10: missing link 'character(1L)'
Rd warning: C:/Users/Manuel/AppData/Local/Temp/RtmpyM0oij/R.INSTALLa6cc51826981/vscDebugger/man/isPromise.Rd:14: missing link 'logical(1L)'
```

and clicking on these links in the documentation just shows
```
Server error: invalid response from R
```

Reply:
We use `Roxygen: list(markdown = TRUE)` in `DESCRIPTION` which means that we need to write documentation in markdown. I don't get the error but it is obvious that these docs are still latex rather than markdown.

Reply:
Indeed, I did not realize that entry in the DESCRIPTION file. I personally never use markdown in roxygen doc - the latex code to remember and type is so minimal in the standard use case that it is not worth bringing markdown to the picture.